### PR TITLE
[MLIRConverter] add nested struct conversion

### PR DESCRIPTION
**Pull request by webmiche**
This PR adds the necessary functionality to convert nested structs form xdsl to MLIR.

### CODE REVIEW

**Review by math-fehr — APPROVED** [review]
I added two nit comments, but otherwise this looks good to me!

File: src/xdsl/mlir_converter.py [hunk 1]
```diff
@@ -39,6 +39,25 @@ def convert_function_type(self, typ: FunctionType) -> mlir.ir.FunctionType:
...
+        from xdsl.printer import Printer
+        from io import StringIO
+        types = []
+        for t in typ.types.data:
+            if isinstance(t, LLVMStructType):
+                types.append(self.get_mlir_struct_string(t))
+            else:
+                file = StringIO("")
+                p = Printer(stream=file)
+                p.print_attribute(t)
```
[the remark this answers is not in the document]
Comment (webmiche) [reply src/xdsl/mlir_converter.py]: Yes you are right. It seems that print internally just calls `str` on the variable it prints. I was able to leverage that to rewrite this loop into a much cleaner form.

[the remark this answers is not in the document]
Comment (math-fehr) [reply src/xdsl/mlir_converter.py]: Oh even better ;)

**Comment by math-fehr** [comment]
Looks good to me, I think we can merge it if it's good for you!

**Pull request merged**